aten/src/ATen/nn_parse.py

```diff
@@ -418,12 +415,22 @@ def run(paths):
                 if cname + suffix in header_functions:
                     bwd_functions.append(header_functions[cname + suffix])
 
-            base = base_declaration(func, fwd_function, backends)
-            declarations.append(forward_declaration(base, fwd_function))
-            declarations.append(backward_declaration(base, bwd_functions))
+            default_scalar_types = ['Float', 'Double', 'Half']  # Half will be stripped for CPU backend
+            forward_backend_types = {}
+            backward_backend_types = {}
+            for backend in backends:
+                backend_props = func.get(backend, {})
+                forward_backend_types[backend] = backend_props.get('forward_scalar_types', default_scalar_types)
+                backward_backend_types[backend] = backend_props.get('backward_backend_types', default_scalar_types)
```

Comment:
As I commented above, I don't see 'backward_backend_types' used anywhere but this.

Reply:
Fixed